packages/@jsii/go-runtime/jsii-experimental/api.go

```diff
@@ -6,22 +6,286 @@ import (
 )
 
 // JSII Kernel API
-// Requests
+
+// Override is a public interface implementing a private method `isOverride`
+// implemented by the private custom type `override`. This is embedded by
+// MethodOverride and PropertyOverride to simulate the union type of Override =
+// MethodOverride | PropertyOverride.
+type Override interface {
+	isOverride()
+}
+
+type override struct{}
+
+func (o override) isOverride() {
+	return
+}
+
+type MethodOverride struct {
+	override
+
+	Method *string `json:"method"`
+	Cookie *string `json:"cookie"`
+}
+
+type PropertyOverride struct {
+	override
+
+	Property *string
+	Cookie   *string
+}
+
+func isMethodOverride(value Override) bool {
+	_, ok := value.(MethodOverride)
+	return ok
+}
+
+func isPropertyOverride(value Override) bool {
+	_, ok := value.(PropertyOverride)
+	return ok
+}
+
+// KernelRequest and KernelResponse allow creating a union of KernelRequest and
+// KernelResponse types by defining private method implemented by a private
+// custom type, which is embedded in all relevant types.
+type KernelRequest interface {
+	req()
+}
```

Comment:
for consistency may want to follow similar pattern as `Override` with `isRequest`/`isResponse`.